test/sensitivities/array.jl

```diff
@@ -1,4 +1,11 @@
-@testset "array" begin
+@testset "sensitivities/array" begin
     @test size(Leaf(Tape(), ones(1, 2, 3, 4))) == (1, 2, 3, 4)
     @test length(Leaf(Tape(), 1:3)) == 3
+
+    let c_rel = 1e4, ε_abs = 1e-16, rng = MersenneTwister(123456)
+        x = randn(2, 10)
+        A = randn(4, 5)
+        f = x̂ -> reshape(x̂, 5, 4) * A
```

Comment:
Why multiply by `A` here?

Reply:
No particular reason; just to generate a slightly realistic test case. I could leave that out if you'd prefer that.

Reply:
The thing that concerns me slightly about this is that this test also depends upon matrix multiplication working as intended, so it's not the case that this unit test passes iff `reshape` is implemented correctly. Roughly speaking, "This unit test passes" implies "correct implementation of `reshape`", but the reverse implication doesn't hold. Practically I don't think is too much of a problem as it seems unlikely that we'll ever revert to a point where matrix-multiplication doesn't work.

To be honest, I'm not too fussed in this case.

Reply:
That's a very reasonable argument; I'll adjust the test accordingly.